### PR TITLE
Manacher's Algo added by wizeewig

### PR DESCRIPTION
**Pull request by wizeewig**
Added the full documentation for Manacher's Algorithm.

### CODE REVIEW

File: tutorials/strings/manachers-algorithm.md [hunk 1]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+
+Given a string, find the longest substring which is palindrome. 
+
+* if the given string is “abaaba”, the output should be “abaaba”
+* if the given string is “abababa”, the output should be “abababa”
+* if the given string is “abcbabcbabcba”, the output should be “abcbabcbabcba”
+
+* ## Brute force Approach O(N³) time
+
+```C++
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:25]: use cpp instead

File: tutorials/strings/manachers-algorithm.md [hunk 2]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+
+# Manacher's Algorithm
+
+There are many ways to find the `longest palindromic substring`. One can find it in O(N³) and also in O(N²) time complexity. But using `Manacher's Algorithm` we do it in `O(N) time`. This algorithm was discovered by `Glenn K. Manacher` in 1975.
+
+Given a string, find the longest substring which is palindrome. 
+
+* if the given string is “abaaba”, the output should be “abaaba”
+* if the given string is “abababa”, the output should be “abababa”
+* if the given string is “abcbabcbabcba”, the output should be “abcbabcbabcba”
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:21]: 1. don't copy from GFG. 
2. give a brief context on what palindrome is
3. besides you should include one that the longest palindromic substring is not same as the original string. i.e. the one you deleted from that site.

File: tutorials/strings/manachers-algorithm.md [hunk 3]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+```
+### Input: 
+daabddfddbegtd
+### Output: 
+The Longest Palindromic Substring is: bddfddb
+### Time complexity O(N³)
+O(N³)
+In the brute force approach, three nested loops are used to find the longest palindromic substring, so the time complexity will be O(N³).
+### Space complexity O(1)
+No extra space is needed in this approach, so the space complexity will be O(1).
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:67]: some line breaks

File: tutorials/strings/manachers-algorithm.md [hunk 4]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+    cout << "The Longest Palindromic Substring is: ";
+    for (int i = start; i <= start+maxLength-1; i++)
+        cout << str[i];
+}
+```
+### Input: 
+daabddfddbegtd
+### Output: 
+The Longest Palindromic Substring is: bddfddb
+### Time complexity O(N³)
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:63]: remove `O(N³)` from the title

File: tutorials/strings/manachers-algorithm.md [hunk 5]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+O(N³)
+In the brute force approach, three nested loops are used to find the longest palindromic substring, so the time complexity will be O(N³).
+### Space complexity O(1)
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:66]: remove `O(1)` from the title

File: tutorials/strings/manachers-algorithm.md [hunk 6]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+No extra space is needed in this approach, so the space complexity will be O(1).
+
+* ## Manacher's Algorithm
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:69]: unnecessary `*`

File: tutorials/strings/manachers-algorithm.md [hunk 7]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+This algorithm is faster than the brute force approach, as it exploits the idea of a palindrome happening inside another palindrome.
+
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:71]: The content starting from this line is copied from https://www.scaler.com/topics/data-structures/manachers-algorithm/. What's the point of copying other content exactly?

File: tutorials/strings/manachers-algorithm.md [hunk 8]
```diff
@@ -9,4 +9,174 @@ keywords:
...
+Each successful comparison results in maxRight moving one step forward, and it never reduces, therefore, the inner while loop gets executed at most n times. Hence, the time complexity of Manacher's algorithm will be O(N).
+
+### Space Complexity
+O(N)
+
+We need O(n) space to create and form p (palindrome width).
+# Author
+
+- [@wizeewig](https://www.github.com/wizeewig)
```
Comment (wingkwong) [inline tutorials/strings/manachers-algorithm.md:181]: Remove this. Use the following. See other tutorials for reference.

```
<TutorialAuthors names="@xxxxxx"/>
```

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. Your changes break the site. Please run the app locally instead of just writing on a markdown editor.

image

2. Use LaTex. e.g. `O(N³)` -> `$O(N ^ 3)$ so it renders like $O(N ^ 3)$ instead of O(N³).

3. I found that most content is just copied from other sites like GFG and Scaler. Please don't do that.

**Comment by wizeewig** [comment]
Ok. Will do the changes.

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. Proofreading first
2. There is not point to duplicate the page title at L14. It's better to call Overview. 
3. I don't see you walk through the algo using LC problems 
4. I don't see the suggested problems at the end
5. Apply LaText on time / space complexity or even those string instead of using `"`. e.g. "abigsquare" -> $abigsquare$

**Comment by wizeewig** [comment]
Ok.

**Review by wingkwong — CHANGES_REQUESTED** [review]
https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorials

**Comment by wingkwong** [comment]
any update?

**Comment by wingkwong** [comment]
ref: #71

**Comment by abhiramtilakiiit** [comment]
I might take this up. Also what syntax checker should I use?

**Comment by wingkwong** [comment]
@abhiramtilakiiit leave a comment in above issue so that I could assign you.

**Pull request closed without merging**